### DISCUSSION TITLE
SRF with DASimpleFoam

### DISCUSSION THREAD

**Discussion by raisePiginYourhome** [mdolab/dafoam#478] [General]
I try the local openfoam for simulation, everything Ok,
when i use DASimpleFoam, then show:
/***
smoothSolver:  Solving for omega, Initial residual = 0.999965, Final residual = 0.00654199, No Iterations 2
smoothSolver:  Solving for k, Initial residual = 0.358909, Final residual = 2.92219e-07, No Iterations 2
smoothSolver:  Solving for Ux, Initial residual = 0.178984, Final residual = 0.00728644, No Iterations 16
smoothSolver:  Solving for Uy, Initial residual = 0.270002, Final residual = 0.0104317, No Iterations 12
smoothSolver:  Solving for Uz, Initial residual = 0.149539, Final residual = 0.00654061, No Iterations 20
GAMG:  Solving for p, Initial residual = 1, Final residual = nan, No Iterations 1000
GAMG:  Solving for p, Initial residual = nan, Final residual = nan, No Iterations 1000

--> FOAM FATAL IO ERROR:
Wrong token type - expected scalar value, found on line 0: word 'nan'

file: /home/dafoamuser/mount/system/data.solverPerformance.p at line 0.

    From function Foam::Istream& Foam::operator>>(Foam::Istream&, Foam::doubleScalar&)
    in file lnInclude/Scalar.C at line 149.
***/

So what's the problem?

**Comment by raisePiginYourhome** [comment]
by the way here is the mesh check result:
`Starting time loop

AMI: Creating addressing and weights between 728 source faces and 728 target faces
AMI: Patch source sum(weights) min:0.999933 max:1 average:0.999998
AMI: Patch target sum(weights) min:0.999946 max:1.00001 average:0.999998
Checking mesh quality for time = 0
    Overall domain bounding box (-0.00673688 -0.0104932 -0.043232) (0.084377 0.0979997 1.50608e-06)
    Mesh has 3 geometric (non-empty/wedge) directions 3{1}
    Mesh has 3 solution (non-empty) directions 3{1}
    Boundary openness (-4.41321e-16 -6.4901e-17 4.60116e-16) OK.
    Max cell openness = 5.23546e-15 OK.
    Max aspect ratio = 232.999 OK.
    Minimum face area = 3.00759e-10. Maximum face area = 3.78033e-05.  Face area magnitudes OK.
    Min volume = 1.15411e-13. Max volume = 1.07269e-07.  Total volume = 7.01154e-05.  Cell volumes OK.
    Mesh non-orthogonality Max: 75.7115 average: 27.1759
    Non-orthogonality check OK.
    Face pyramids OK.
    Max skewness = 2.32819 OK.
    Coupled point location match (average 0) OK.

Mesh OK.`

**Comment by friedenhe** [comment]
Can you run `renumberMesh -overwrite` and then run DASimpleFoam again? Also, can you try the SA model?

**Reply by raisePiginYourhome** [reply]
it still not work, when i use SA model in running local openfoam SRFSimpleFoam, the "Floating point exception" shown.
I modified the divSchemes, make it could run with K-Omega RANS model, but the values of objFunc seem incorrect.

![value_show](https://github.com/mdolab/dafoam/assets/126868299/d9f86d85-7e75-4e00-8184-e62bbce24dd0)

But in the local Openfoam, the pressure of inlet should be -839.327kpa and outlet is 101.63kpa.
Is there any way to solve this, hope you can illustrate the principles behind this.

**Comment by raisePiginYourhome** [comment]
Warning: Cutting back alpha due to evaluation error
opt_IPOPT file has this error, how to solve this?

**Comment by friedenhe** [comment]
> it still not work, when i use SA model in running local openfoam SRFSimpleFoam, the "Floating point exception" shown.

I am confused, did you run SRFSimpleFoam or DAFoam? DAFoam does not support SRF, it supports only MRF. The BCs between these two are different..

**Reply by raisePiginYourhome** [reply]
I run the SRFSimpleFoam.

**Comment by friedenhe** [comment]
> Warning: Cutting back alpha due to evaluation error opt_IPOPT file has this error, how to solve this?

The message said it was a warning, not an error...

**Reply by raisePiginYourhome** [reply]
OK, thank you very much.

**Reply by raisePiginYourhome** [reply]
What should I do if I want the control points to move larger and the mesh deformation to be more pronounced？

**Reply by raisePiginYourhome** [reply]
`List of user-set options:

                                    Name   Value                used
                             alpha_for_y = full                  yes
                         constr_viol_tol = 1e-05                 yes
                        file_print_level = 5                     yes
                   hessian_approximation = limited-memory        yes
              limited_memory_max_history = 10                    yes
                           linear_solver = mumps                 yes
                                max_iter = 100                   yes
                             mu_strategy = adaptive              yes
                      nlp_scaling_method = none                  yes
                             output_file = opt_IPOPT.txt         yes
                             print_level = 5                     yes
                      print_user_options = yes                   yes
                                recalc_y = yes                   yes
                                      sb = yes                   yes
                                     tol = 1e-05                 yes
This is Ipopt version 3.13.5, running with linear solver mumps.
NOTE: Other linear solvers might be more efficient (see Ipopt documentation).

Number of nonzeros in equality constraint Jacobian...:        0
Number of nonzeros in inequality constraint Jacobian.:       48
Number of nonzeros in Lagrangian Hessian.............:        0

Total number of variables............................:       48
                     variables with only lower bounds:        0
                variables with lower and upper bounds:       48
                     variables with only upper bounds:        0
Total number of equality constraints.................:        0
Total number of inequality constraints...............:        1
        inequality constraints with only lower bounds:        0
   inequality constraints with lower and upper bounds:        0
        inequality constraints with only upper bounds:        0

iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
   0  4.6694305e+00 0.00e+00 5.68e-02   0.0 0.00e+00    -  0.00e+00 0.00e+00   0
Warning: Cutting back alpha due to evaluation error
Warning: Cutting back alpha due to evaluation error
   1  4.6273602e+00 0.00e+00 5.56e-02  -2.7 5.67e-02    -  9.90e-01 2.50e-01f  3
   2  4.5817142e+00 0.00e+00 5.48e-02  -0.8 3.67e+00    -  9.96e-01 2.38e-07f 23
   3  4.5811318e+00 0.00e+00 1.33e-01  -6.6 1.17e-01    -  9.90e-01 2.38e-07f 23
   4  4.5875913e+00 0.00e+00 7.73e-02  -4.5 2.74e-01    -  1.00e+00 3.16e-322h1069
Warning: Cutting back alpha due to evaluation error
   5  4.5783096e+00 0.00e+00 5.49e-02  -4.6 5.02e-02    -  1.00e+00 2.50e-01f  3
   6  4.5776097e+00 0.00e+00 5.47e-02  -6.5 5.49e-02    -  1.00e+00 7.63e-06f 18
   7  4.6458197e+00 0.00e+00 1.10e-01  -8.4 6.32e-04    -  1.00e+00 6.63e-316h1048
   8  4.5773159e+00 0.00e+00 5.52e-02  -8.5 6.84e-04    -  1.00e+00 2.50e-01f  3
   9  4.6464202e+00 0.00e+00 5.63e-02 -10.4 1.80e-02    -  1.00e+00 3.24e-319h1059
Warning: Cutting back alpha due to evaluation error
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
  10  4.5868143e+00 0.00e+00 1.13e-01 -10.5 1.84e-02    -  1.00e+00 6.25e-02f  5
  11  4.5816679e+00 0.00e+00 6.99e-02 -10.5 2.86e+01    -  1.00e+00 2.38e-07f 23
  12  4.5814571e+00 0.00e+00 5.52e-02 -10.5 9.34e-03    -  1.00e+00 3.81e-06f 19
  13  4.5795786e+00 0.00e+00 5.08e+00 -10.5 9.44e-03    -  1.00e+00 1.25e-01f  4
  14  4.5763831e+00 0.00e+00 5.50e-02 -10.5 7.25e-02    -  1.00e+00 7.81e-03f  8
  15  4.5880204e+00 0.00e+00 1.27e+01 -11.0 5.50e-02    -  1.00e+00 2.02e-320h1063
  16  4.5816455e+00 0.00e+00 9.83e-01 -11.0 1.27e+01    -  1.00e+00 1.16e-10f 34
  17  4.5816357e+00 0.00e+00 7.50e-02 -11.0 1.97e-09    -  1.00e+00 6.10e-05f 15
  18  4.6607222e+00 0.00e+00 5.65e-02 -11.0 6.27e-10    -  1.00e+00 4.67e-302h1002
  19  4.6583628e+00 0.00e+00 1.96e-01 -11.0 6.26e-10    -  1.00e+00 5.00e-01f  2
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
  20  4.6454421e+00 0.00e+00 2.51e-01 -11.0 7.49e-10    -  1.00e+00 1.00e+00f  1
  21  4.6094970e+00 0.00e+00 5.40e-02 -11.0 2.51e-01    -  1.00e+00 1.25e-01f  4
  22  4.6009022e+00 0.00e+00 5.49e-02 -11.0 9.84e-03    -  1.00e+00 3.91e-03f  9
  23  4.5995192e+00 0.00e+00 1.66e+00 -11.0 1.03e-02    -  1.00e+00 9.77e-04f 11
  24  4.5839331e+00 0.00e+00 5.53e-02 -11.0 3.44e-03    -  1.00e+00 2.44e-04f 13
Warning: Evaluation error during soft restoration phase step.
Warning: Evaluation error during soft restoration phase step.
Warning: Evaluation error during soft restoration phase step.
Restoration phase is called at point that is almost feasible,
  with constraint violation 0.000000e+00. Abort.

Number of Iterations....: 24

                                   (scaled)                 (unscaled)
Objective...............:   4.5839331470834228e+00    4.5839331470834228e+00
Dual infeasibility......:   5.5267682859370941e-02    5.5267682859370941e-02
Constraint violation....:   0.0000000000000000e+00    0.0000000000000000e+00
Complementarity.........:   1.0000017180366254e-11    1.0000017180366254e-11
Overall NLP error.......:   5.5267682859370941e-02    5.5267682859370941e-02


Number of objective function evaluations             = 6569
Number of objective gradient evaluations             = 25
Number of equality constraint evaluations            = 0
Number of inequality constraint evaluations          = 6566
Number of equality constraint Jacobian evaluations   = 0
Number of inequality constraint Jacobian evaluations = 25
Number of Lagrangian Hessian evaluations             = 0
Total CPU secs in IPOPT (w/o function evaluations)   =      0.647
Total CPU secs in NLP function evaluations           =  22680.907

EXIT: Restoration Failed!
`

**Reply by raisePiginYourhome** [reply]
It seems the optimization results not very good, is there any options to imporve this situation?

**Reply by friedenhe** [reply]
Try to use the SLSQP optimizer.

**Reply by raisePiginYourhome** [reply]
thanks for your assistance, I'll try it.
